src/web/views/Content/Select/index.jsx

```diff
@@ -0,0 +1,130 @@
+  fetchContentMessagesCount,
+  fetchContentCategories,
+  upsertContentMessages
+} from '~/actions'
+
+const style = require('./style.scss')
+
+class SelectContent extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { show: false, category: null, searchTerm: '', contentToEdit: null }
+
+    window.botpress = window.botpress || {}
+    window.botpress.pickContent = (options = {}, callback) => {
+      this.props.fetchContentMessagesRecent({})
+      this.props.fetchContentMessagesCount()
+      this.props.fetchContentCategories()
+      this.setState({ show: true, callback })
+    }
+  }
+
+  componentWillUnmount() {
+    delete window.botpress.pickContent
+  }
+
+  search = event => {
+    this.setState({ searchTerm: event.target.value })
+    this.props.fetchContentMessagesRecent({ searchTerm: event.target.value })
+  }
+
+  handleCreate = () => {
+    this.props
+      .upsertContentMessages({ categoryId: this.state.category.id, formData: this.state.contentToEdit })
+      .then(() =>
+        Promise.all([
+          this.props.fetchContentMessagesRecent({ searchTerm: this.state.searchTerm }),
+          this.props.fetchContentMessagesCount()
+        ])
+      )
+      .then(() => this.setState({ category: null, contentToEdit: null }))
+  }
+
+  handlePick(item) {
+    this.setState({ show: false })
+    this.state.callback(item)
+  }
+
+  handleFormEdited = data => {
+    this.setState({ contentToEdit: data })
+  }
+
+  render() {
+    const props = this.props
+    const noop = () => {}
+
+    const onClose = () => this.setState({ show: false })
+    const onSubmit = () => this.setState({ show: false })
+
+    const schema = (this.state.category || {}).schema || { json: {}, ui: {} }
+
+    return (
+      <Modal animation={false} show={this.state.show} onHide={onClose} container={document.getElementById('app')}>
+        <Modal.Header closeButton>
+          <Modal.Title>Pick Content</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <input
+            type="text"
+            className="form-control"
+            placeholder={`Search all content elements (${this.props.messagesCount})`}
+            aria-label="Search content elements"
+            onKeyUp={this.search}
+          />
+          <hr />
+          <div className="list-group">
+            {this.props.categories.map(category => (
+              <a
+                href="#"
+                onClick={() => this.setState({ category, contentToEdit: {} })}
+                className={`list-group-item list-group-item-action ${classnames(style.createItem)}`}
+              >
+                Create new {category.title}
+              </a>
+            ))}
+            {this.props.messages.map(message => (
+              <a href="#" className="list-group-item list-group-item-action" onClick={() => this.handlePick(message)}>
+                {`[${message.categoryId}] ${message.data && message.data.question}`}
```

Comment:
Not sure where are these data coming from, but why is the field called `question`? Isn't it specific to trivia? What we probably want is the `previewText`?
Can as well just misunderstand what these messages are.

Reply:
Agreed, this is trivia specific. Needs to be `previewText`

Reply:
Done!